fili-core/src/main/java/com/yahoo/bard/webservice/metadata/SegmentMetadata.java

```diff
@@ -92,9 +80,13 @@ private SegmentMetadata(Map<String, Set<Interval>> dimensionIntervals, Map<Strin
         return metricIntervals;
     }
 
+    public boolean isEmpty() {
+        return isEmpty;
+    }
+
```

Comment:
Was there a rationale for calculating this at construction time rather than using the expression here?